std/experimental/perpetual.d

```diff
@@ -0,0 +1,653 @@
+module std.experimental.perpetual;
+/** TODO: 
+ *		documentation
+ *		version(Windows) impl
+ */
+
+private import std.exception;
+private import std.conv : to, emplace;
+private import std.file : exists;
+private import std.traits : hasIndirections, isCallable, isAssignable, isArray;
+
+/**
+ * Creates an object mapped to a file. The object is persistent 
+ *   and outlives the parent application. The file may be reopened
+ *   and the object reused, it's value persists. This might be viewed
+ *   as kind of binary serialization with the difference that access
+ *   to the variable is almost as fast as any regular memory. The 
+ *   structure might be also used as memory shared between processes/threads,
+ *   however, no built-in synchronization is provided.
+ *
+ * If the template parameter T is immutable, the created object is also
+ *   immutable. Note: shared memory is then created in read-only
+ *   mode, so even cast to mutable and attempt to modify it will cause
+ *   segmentation fault.
+ *
+ * The Perpetual(T) exists in two forms. When T is regular value type
+ *   (including static arrays), Persistent(T) behaves like reference to
+ *   the object. In the second form, Persistent(T[]), the class behaves
+ *   as proxy to dynamic array providing slicing interface. Note, even
+ *   in this case, elements of the array must be value type, no pointers
+ *   nor references to process's memory allowed.
+ *
+ * The values created are initialized with std.conv.emplace using
+ *  additional arguments if any. The object remains in valid state
+ *  until the file is deleted or modified.
+ **/
+struct Perpetual(T)
+{
+	// underlying shared memory allocator
+	private ShMem _heap;
+	private enum _tag="Perpetual!("~T.stringof~")";
+
+	//dynamic array
+	static if(is(T == Element[],Element))
+	{
+		private Element[] _value;
+		private enum bool dynamic=true;
+		static assert(!hasIndirections!Element
+		    , Element.stringof~" is reference type");
+
+		/** Proxy to underlying dynamic array type.
+		 *  Forwards calls to opSlice(), length() etc
+		 */
+		@property Element[] Ref() { return _value; }
+
+		/// Convert to string the value, not the class itself
+		@property toString() { return to!string(_value); }
```

Comment:
This code unnecessarily allocates. Please fix by using the overload shown in http://dlang.org/phobos/std_format.html#.formatValue.10

This applies to all `toString()` functions in this code.